app/assets/javascripts/discourse/app/components/sidebar/edit-navigation-menu/categories-modal.gjs

```diff
@@ -18,126 +18,148 @@ import gt from "truth-helpers/helpers/gt";
 import includes from "truth-helpers/helpers/includes";
 import not from "truth-helpers/helpers/not";
 
+function uniqBy(elements, f) {
```

Comment:
This feels unnecessary, I see `uniqBy` and `uniq` as available methods for arrays in JS. I suspect one of those can be helpful